tests/TestCase/Database/QueryTest.php

```diff
@@ -4221,11 +4276,15 @@ public function testSqlCaseStatement()
             'Not published',
             'None'
         ];
+        $finalExpression = $query->newExpr();
+        $this->withErrorReporting(E_ALL ^ E_USER_DEPRECATED, function () use ($conditions, $values, $finalExpression) {
+            $finalExpression->addCase($conditions, $values);
+        });
```

Comment:
I don't know if this is the best way to test for deprecated behavior. Should I change this?

Reply:
We have a `$this->deprecated (function () { ... });` helper method for toggling error reporting allowing deprecated methods to be called in tests.